ComSemApp/utils.py

```diff
@@ -70,12 +74,62 @@ def transcribe(request):
             try:
                 text = r.recognize_google(audio)
                 #closes the temp files
-                os.close(in_file_handle)
-                os.close(out_file_handle)
+                close(in_file_handle)
+                close(out_file_handle)
                 # capitalize sentence
                 return HttpResponse(text.capitalize())
             
             except Exception:
-                os.close(in_file_handle)
-                os.close(out_file_handle)
+                close(in_file_handle)
+                close(out_file_handle)
                 return HttpResponse("")
+
+def get_youglish_videos(request : HttpRequest) -> HttpResponse:
+    """
+        Polls YouGlish REST API for YouTube video clips containing the phrase given in an HTTP GET request
+
+        Arguments:
+            request : HttpRequest - a Django HttpRequest object which should contain GET request data
+                phrase (required) - The phrase to search for
+                accent (optional) - A code which allows the client to search for a particular accent
+                page   (optional) - Used for pagination to get more results
+        
+        Returns:
+            HttpResponse - A Django HttpResponse object indicating the outcome of the request:
+                JsonResponse           (200) - A success message containing the requested data in JSON format
+                HttpResponseBadRequst  (400) - If the request does not have the required GET arguments
+                Http404                (404) - If the requested phrase has no available video clips
+                HttpResponeServerError (500) - If some problem occurs in communicating with YouGlish
+
```

Comment:
add into this the format of the json response data

Reply:
The api is _technically_ subject to change, so I don't want to codify in our codebase and instead link to it